### DISCUSSION TITLE
Confusing errors reported by Post Harden Runner

### DISCUSSION THREAD

**Discussion by jmgrady** [step-security/harden-runner#230] [Q&A]
I am integrating the Harden Runner action into the CI/CD GitHub Actions for https://github.com/sillsdev/TheCombine.  I have followed the recommendations made by the audit but there are two errors that I see in the Post Harden Runner stage that I do not expect.  All of the actions report:
```
StepSecurity Harden Runner: Overwrite detected for /etc/systemd/resolved.conf
```
It makes sense that resolved.conf would be overwritten so I am surprised that it shows up as an error.

If the action builds a Docker container, then I also see this error:
```
StepSecurity Harden Runner: DNS resolution for domain docker.io. was blocked. This domain is not in the list of allowed-endpoints.
```
The build seems to have completed successfully, however.

In addition, the GitHub Action passes so it seems that these should be "info" or "warning" messages.

Is there something I need to do to address these messages?

**Comment by varunsh-coder** [comment]
Hi @jmgrady, thanks for creating this discussion and using harden runner!

w.r.t `StepSecurity Harden Runner: Overwrite detected for /etc/systemd/resolved.conf` this might be a bug where harden runner is detecting its change of the resolved.conf. I don't see any workflow steps that might change this file. I will look into it more. This will likely need a bug fix in harden runner. 

w.r.t `DNS resolution for domain docker.io. was blocked` this is odd. Typically after DNS resolution, the code would call to this domain. But that call is not showing up in the report. Let me check if something like this is showing in other repositories. 

I will get back to you after investigating these. As of now, you don't need to do anything to address the error messages.

**Reply by jmgrady** [reply]
Thanks for your quick reply! I look forward to hearing about what you find.

**Comment by varunsh-coder** [comment]
@h0x0er, can you please investigate these?

**Comment by varunsh-coder** [comment]
The `DNS resolution for domain docker.io. was blocked` issue was resolved in release v2.2.0. 

We are still investigating the other issue.

**Comment by varunsh-coder** [comment]
@jmgrady all the issues should be resolved now. Please let me know if you see any of these issues in the recent runs. Thanks!
